BE/app/domain/auth/auth_repository.py

```diff
@@ -1,16 +1,20 @@
 from typing import Optional
 
+from app.domains.helpers.database_repository import DatabaseRepository
 from app.model.auth import Auth
-from app.infrastructure.postgresql.database import session
 from app.infrastructure.postgresql.user.user_dto import UserDTO
 
 
 class AuthRepository:
     """User Repository defines a repository interface for user entity."""
 
+    def __init__(self, database_repository: DatabaseRepository):
+        self.db_repo = database_repository
+        self.db = next(self.db_repo.get_db())
```

Comment:
anh @bao-qnguyen97
I used it here

Reply:
you can apply DatabaseRepository as the parent class of AuthRepository